src/Abstractions/NexusMods.Abstractions.Settings/SingleValueMultipleChoiceContainer.cs

```diff
@@ -0,0 +1,30 @@
+using JetBrains.Annotations;
+
+namespace NexusMods.Abstractions.Settings;
+
+/// <summary>
+/// Container for a singe value with multiple choices.
+/// </summary>
+[PublicAPI]
+public sealed class SingleValueMultipleChoiceContainer : APropertyValueContainer<object>
+{
+    private readonly object[] _allowedValues;
+    private readonly Func<object, string> _valueToTranslation;
```

Comment:
Maybe rename `_valueToTranslation` to `_valueToString` or something more generic like that